java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java

```diff
@@ -384,17 +388,91 @@ private void validatePublicationConfig(boolean isSuperUser) throws SQLException
+            }
+        }
+        //  Only after v13, PG allows adding a partitioned table to a publication. So, if the
+        // version is before v13, the tables in a publication are always partition leaves, we don't
+        // check their ancestors and descendants anymore.
+        if (isPublicationExists && pgVersion >= 13) {
+            List<String> family = new ArrayList<>();
+            boolean findRoot = false;
+            String currentPartition = tableName;
+            while (!findRoot) {
+                try (var stmt =
+                        jdbcConnection.prepareStatement(
+                                ValidatorUtils.getSql("postgres.partition_parent"))) {
+                    stmt.setString(
+                            1, String.format("\"%s\".\"%s\"", this.schemaName, currentPartition));
+                    stmt.setString(
+                            2, String.format("\"%s\".\"%s\"", this.schemaName, currentPartition));
+                    stmt.setString(
+                            3, String.format("\"%s\".\"%s\"", this.schemaName, currentPartition));
+                    var res = stmt.executeQuery();
+                    if (res.next()) {
+                        String parent = res.getString(1);
+                        family.add(parent);
+                        currentPartition = parent;
+                    } else {
+                        findRoot = true;
+                    }
+                }
+            }
+            try (var stmt =
+                    jdbcConnection.prepareStatement(
+                            ValidatorUtils.getSql("postgres.partition_descendants"))) {
+                stmt.setString(1, String.format("\"%s\".\"%s\"", this.schemaName, this.tableName));
+                stmt.setString(2, String.format("\"%s\".\"%s\"", this.schemaName, this.tableName));
+                var res = stmt.executeQuery();
+                while (res.next()) {
+                    String descendant = res.getString(1);
+                    family.add(descendant);
+                }
+            }
+
+            for (String relative : family) {
+                try (var stmt =
+                        jdbcConnection.prepareStatement(
+                                ValidatorUtils.getSql("postgres.partition_in_publication.check"))) {
+                    stmt.setString(1, schemaName);
+                    stmt.setString(2, relative);
+                    stmt.setString(3, pubName);
+                    var res = stmt.executeQuery();
+                    while (res.next()) {
+                        if (res.getBoolean(1)) {
+                            throw ValidatorUtils.invalidArgument(
+                                    String.format(
+                                            "The ancestor or descendant partition '%s' of the table partition '%s' is already covered in the publication '%s'. Please use a new publication for '%s'",
```

Comment:
May add a comment for the link to the reference of this limitation.

Reply:
No good reference found. Added more comments to explain.